shutdown.go

```diff
@@ -74,10 +105,61 @@ func (app *App) broadcastSignal(signal os.Signal) error {
 	}
 
 	if unsent != 0 {
-		return fmt.Errorf("failed to send %v signal to %v out of %v channels",
-			signal, unsent, len(app.dones),
-		)
+		return ErrOnUnsentSignal{
+			Signal:   signal,
+			Unsent:   unsent,
+			Channels: len(app.dones),
+		}
 	}
 
 	return nil
 }
+
+func (app *App) broadcastWaitSignal(signal os.Signal, code int) error {
+	app.waitsMu.Lock()
+	defer app.waitsMu.Unlock()
+
+	app.shutdownSignal = &ShutdownSignal{
+		Signal:   signal,
+		ExitCode: code,
+	}
+
+	var unsent int
+	for _, wait := range app.waits {
+		select {
+		case wait <- *app.shutdownSignal:
+		default:
+			// shutdown called when wait channel has already received a
+			// termination signal that has not been cleared
+			unsent++
+		}
+	}
+
+	if unsent != 0 {
+		return ErrOnUnsentSignal{
+			Signal:   signal,
+			Unsent:   unsent,
+			Code:     code,
+			Channels: len(app.waits),
+		}
+	}
+
+	return nil
+}
+
+// ErrOnUnsentSignal ... TBD
+type ErrOnUnsentSignal struct {
```

Comment:
Is there a good use case for users to want to match on this error?
If not, we can keep it unexported (errUnsentSignal).
(If we want to export it, we can call it ErrUnsentSignal.)

Reply:
It was just a guess on my part, assuming people want to know the signal and exit code information to handle it, as opposed to some other error type. It's probably wiser to just make it private however, and if someone raises the use-case we can make it public then.